packages/core/admin/admin/src/types/adminAPI.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable check-file/filename-naming-convention */
+
+export interface APIBaseParams {
+  filter?: unknown;
+  locale?: unknown;
+  publicationState?: unknown;
+  sort?: unknown;
+}
```

Comment:
I think this is enough for now, but if we wanted to type it a bit more we could do:

```ts
import { Strapi, EntityService, Common } from '@strapi/types';

type Filters = EntityService.Params.Filters.Any<Common.UID.Schema>
type Sort = EntityService.Params.Sort.Any<Common.UID.Schema>

export interface APIBaseParams {
  filters?: Filters;
  locale?: string;
  publicationState?: 'preview' | 'live';
  sort?: Sort;
}
```

also, I think the param to filter is `filters` (not checked though)

Reply:
What I don't like here is referencing the Entity Service in the UI, I think we should add an aliasing somewhere else (shared folder pbbly)

Reply:
I think that would make sense, since the filters (for example) for the ES and the API would differ when it comes to everything else `string` types, because dates, id's ... all become strings in the end.

In between there is qs, which converts everything else, so there is no 1:1 translation from the ES.

Reply:
true! This generic filter wouldn't know anything about the schema, but only ensure the filter contract is a valid one.
The next step would involve the schema mapping you mention, taking into consideration the type translation. And that sounds challenging

Reply:
I've:

- renamed `filter` to `filters`
- added your suggestion for `publicationState`
- added your suggestion for `locale`

I think I'll leave the ES types out for the moment until we have a general direction here.

Reply:
sounds good 🚀